### DISCUSSION TITLE
0.1.1 Pre-Release

### DISCUSSION THREAD

**Discussion by RobertD502** [RobertD502/home-assistant-flair#32] [General]
## 👋 Welcome!
**Use this discussion board to post any potential bugs that you may have found in the `0.1.1` pre-release.**

**Comment by plapin751** [comment]
Thank you for all your hard work

**Comment by karl-vanderslice** [comment]
Thanks as well!  Just installed the new version, oAuth looks great.

**Comment by fcfort** [comment]
Were there changes made to the login flow? Are we supposed to re-do our login when upgrading to this beta version? I see the following error in the logs after upgrading:

`2022-08-22 09:23:33.240 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Flair' for flair integration not ready yet: Flair API error: Invalid Client ID or Secret provided; Retrying in background`

**Reply by fcfort** [reply]
Yep, that was it, I needed to use OAuth 2.0 API credentials. Might be worth noting that in the release notes.

**Reply by RobertD502** [reply]
It is noted in the release notes.

**Reply by fcfort** [reply]
Ah yep, sorry. I also noticed that all my minisplit entities have been renamed, which caused all my existing thermostats in the UI to show as unavailable:

![image](https://user-images.githubusercontent.com/435867/185996481-c62bb071-6c43-4783-9c25-a5507806cee1.png)

**Reply by RobertD502** [reply]
Since the majority of the entities needed to be renamed to follow the naming scheme introduced in 2022.8, I recommended (in the release notes) for users to delete their current integration and set up Flair from scratch.

**Comment by fcfort** [comment]
I noticed the thermostat cards for the mini-splits no longer have an "off" button, e.g.: 

![image](https://user-images.githubusercontent.com/435867/186068943-e7962bd4-aa45-40ad-8958-8742466c0fca.png)

Is this intended? I do see a comment in the release notes: "if your structure is in manual mode, you can turn your mini split on/off by changing the Preset setting to either 'On' or 'Off'.", but I don't understand why we have to go into the entities sub-menu just to be able to turn off the mini-split.

My structure is in manual mode.

**Reply by RobertD502** [reply]
The change reflects the fact that Flair's Power mode is separate from HVAC mode.

If you turn power to the mini split off, it is incorrect to state that the HVAC mode is "off". The mini split remains in the HVAC mode that was set before it was powered off.

If you take a thermostat like the ecobee for example, it has an actual HVAC mode of "Off".

Secondary reason:

For Flair, if you're in manual mode, you can't make any changes to HVAC mode, swing, fan before the mini split is turned on. There is no HVAC mode of "On" within Home Assistant. To accomplish this we could revert back to the original code to send an on command before calling a hvac mode of Heat, cool, etc. However, this has the consequence of user's mini splits turning on when they thought they are just changing the HVAC mode. By requiring users to make sure mini splits are turned on before making changes, we are mirroring the requirements set by Flair and avoiding unintended consequences.

This is also reflected in official Home Assistant integrations such as Sensibo (another mini split controller) - this integration has no power button as Off is not a HVAC mode for mini splits. They actually require users to create their own switches via YAML to control power on and off. I decided to make this easier for users and offer power through the preset options.

**Reply by RobertD502** [reply]
If you want to show presets directly on the card, there are several custom cards out there such as https://github.com/nervetattoo/simple-thermostat#usage-of-the-control-config that can do this.

**Comment by plapin751** [comment]
thanks for that card,  that works pretty nice.   has on/off on the card like @fcfort wanted


![temp](https://user-images.githubusercontent.com/77878008/186078197-e5876ba3-30fe-4004-ab1f-10db5f436dd6.jpg)

**Reply by RobertD502** [reply]
Just want to give you a heads up that 0.1.1 has preset removed as Home Assistant was calling to set hvac mode, fan speed, etc when using preset modes - we can't have this as Flair doesn't allow changing those when the unit is powered off. So, instead, there is a switch entity now dedicated to power.

The card you are using allows you to integrate the switch entity into it. So, be sure to make that change.

**Reply by plapin751** [reply]
thanks for the heads up,  i just tested it with some automations.   works great!

**Reply by plapin751** [reply]
@RobertD502  I've been pretty happy with new release,  been working on this card with my limited skills.
I can post the yaml if anybody wants it. mostly combination of mushroom cards

![Flair](https://user-images.githubusercontent.com/77878008/187810971-0ce97786-f748-47e3-a3e3-24bc25470c80.jpg)

**Reply by worm5406** [reply]
> @RobertD502 I've been pretty happy with new release, been working on this card with my limited skills. I can post the yaml if anybody wants it. mostly combination of mushroom cards
> 
> ![Flair](https://user-images.githubusercontent.com/77878008/187810971-0ce97786-f748-47e3-a3e3-24bc25470c80.jpg)

Sure, post the yaml. I'll take a look at it.

I just upgraded to the new version and I was in the process of a new FLAIR install. It works perfectly and the auth worked as described. 

I have forced air AC and baseboard hot water Heat. Ecobee thermostats.

So I am combining a few items. Now to make some controls for them to work together.

**Reply by plapin751** [reply]
@worm5406 
I am open and thankful for any comments 

https://github.com/plapin751/Flair-Card/blob/main/flair.yml

**Reply by RobertD502** [reply]
> @worm5406 
> 
> I am open and thankful for any comments 
> 
> 
> 
> https://github.com/plapin751/Flair-Card/blob/main/flair.yml

Did you make the repo private? If so, others can't access it.

**Reply by worm5406** [reply]
I can not access it either.

**Reply by plapin751** [reply]
@worm5406 @RobertD502   i made it public now

**Comment by fcfort** [comment]
Changing the preset from "on" to "off" when starting in Drying mode gives the following error in the UI:

`Failed to call service climate/set_hvac_mode. expected HVACMode or one of 'off', 'heat', 'cool', 'heat_cool', 'auto', 'dry', 'fan_only' for dictionary_value @ data['hvac_mode']`

<img src="https://user-images.githubusercontent.com/435867/186289338-3ca9fc9c-eeb9-4bef-85cc-6fc2b2caf160.png" width=50% height=50%>

**Reply by RobertD502** [reply]
Looks like this occurs occasionally when the call to Flair's servers during the refresh doesn't return the HVAC mode. However, from my testing this seems to be intermittent, but the good news is that the preset "off" service call completes successfully. The "operation" aka HVAC mode gets updated correctly during the next refresh.

Error occurring:
![Preset Error](https://user-images.githubusercontent.com/52541649/186778857-c08cdfc0-797e-45bf-9b34-1f85a1ef6c80.gif)


No Error:
![No Error](https://user-images.githubusercontent.com/52541649/186778884-f52f90b9-8a71-4756-8655-915f928c87b6.gif)


I'll get around this by setting the HVAC mode manually during preset service calls as we always know that the HVAC mode will remain what it was before the preset is called.

**Reply by RobertD502** [reply]
Disregard...it is not on Flair's end at all. I dug a bit deeper and can see the correct mode is returned after calling a preset. The problem is with Home Assistant. With preset modes, HA is set up to make a call to all the other services as well. Not sure why they went with this design. As you know, you can't change the HVAC mode when the mini split is off, which is causing problems as home assistant is designed to call the set function for those whenever a preset is set. 

Looks like I may have to move the power over to a switch like other integrations do instead of preset. Hold tight...going to test out using a switch now.

**Reply by RobertD502** [reply]
Can confirm that using a switch instead of preset gets rid of the problem....I'll be creating another release with the changes.

**Reply by RobertD502** [reply]
0.1.1 has been released with new switch entity for power and preset removed. You'll also notice that setting temperature is enabled when HVAC is off and structure mode is manual....I had to revert to this as the climate entity requires at least one feature to be enabled (the preset mode feature did this before). However, when setting temperature with the HVAC unit off, I log an error letting users know that the unit needs to be On to support setting temperature.
